sympy/calculus/singularities.py

```diff
@@ -32,12 +32,19 @@ def singularities(expr, sym):
     .. [1] http://en.wikipedia.org/wiki/Mathematical_singularity
 
     """
-    if not expr.is_rational_function(sym):
-        raise NotImplementedError("Algorithms finding singularities for"
-                                  " non rational functions are not yet"
-                                  " implemented")
-    else:
-        return solveset(simplify(1/expr), sym)
+    f, sym = map(sympify, [f, sym])
+
+    if not isinstance(sym, Symbol):
+        raise ValueError("")
+    if f.is_number or f.is_polynomial(sym):
+        return S.EmptySet
+    elif f.is_Mul or f.is_Add:
+        return Union(*[singularities(funct, sym) for funct in f.args])
```

Comment:
This is where more refined analysis is necessary. Consider ``singularities(sin(x)/x, x)``.